foam_controlwrapper/tests/test_foam_mesh.py

```diff
@@ -0,0 +1,270 @@
+        """Test add_point method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        with self.assertRaises(NotImplementedError):
+            foam_mesh.add_point(self.points[4])
+
+    def test_add_edge(self):
+        """Test add_edge method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        with self.assertRaises(NotImplementedError):
+            foam_mesh.add_edge(self.edge)
+
+    def test_add_face(self):
+        """Test add_face method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        with self.assertRaises(NotImplementedError):
+            foam_mesh.add_face(self.faces[4])
+
+    def test_add_cell(self):
+        """Test add_cell method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        with self.assertRaises(NotImplementedError):
+            foam_mesh.add_cell(self.cells[0])
+
+    def test_update_point(self):
+        """Test update_point method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        with self.assertRaises(NotImplementedError):
+            foam_mesh.update_point(self.points[4])
+
+    def test_update_edge(self):
+        """Test update_edge method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        with self.assertRaises(NotImplementedError):
+            foam_mesh.update_edge(self.edge)
+
+    def test_update_face(self):
+        """Test update_face method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        with self.assertRaises(NotImplementedError):
+            foam_mesh.update_face(self.faces[4])
+
+    def test_update_cell(self):
+        """Test update_cell method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        cell = self.cells[0]
+        cell.data[CUBA.VELOCITY] = (2, 1, 3)
+        foam_mesh.update_cell(cell)
+        cell_f = foam_mesh.get_cell(cell.uid)
+        self.assertEqual(cell.data[CUBA.VELOCITY], cell_f.data[CUBA.VELOCITY])
+
+        cell.points = [self.points[1].uid, self.points[2].uid,
+                       self.points[3].uid]
+        with self.assertRaises(Warning):
+            foam_mesh.update_cell(cell)
+
+    def test_iter_points(self):
+        """Test iter_points method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        puids_f = [point.uid for point in foam_mesh.iter_points()]
+        self.assertEqual(set(self.puids), set(puids_f))
+
+    def test_iter_edges(self):
+        """Test iter_edges method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        self.assertEqual(foam_mesh.iter_edges(), [])
+
+    def test_iter_faces(self):
+        """Test iter_faces method
+
+        """
+
+        foam_mesh = FoamMesh('test_mesh', self.mesh)
+        for face_f in foam_mesh.iter_faces():
+            face = self.mesh.get_face(face_f.uid)
+            self.assertEqual(face.data[CUBA.LABEL],
```

Comment:
CUBA.LABEL is used for grouping the faces for boundary conditions. We should have in the future sme grouping for mesh objects to get this done in a more elegant and efficient way. This information should also be saved along with the mesh itself.